dot/state/block.go

```diff
@@ -250,6 +251,37 @@ func (bs *BlockState) GetHashByNumber(num uint) (common.Hash, error) {
 	return common.NewHash(bh), nil
 }
 
+// GetHashesByNumber returns the block hashes with the given number
+func (bs *BlockState) GetHashesByNumber(num uint) ([]common.Hash, error) {
+	blockHashes := []common.Hash{}
+	block, err := bs.GetBlockByNumber(num)
+	if err == nil {
+		blockHashes = bs.bt.GetAllBlocksAtNumber(block.Header.ParentHash)
+	}
+
+	hash, err := bs.bt.GetHashByNumber(num)
+	if err == nil {
```

Comment:
Nit: Could be nice to reframe this as if err != nil to stick with normal pattern

Reply:
I agree. Can we switch this around? You can do the `errors.Is` check in a `if err != nil ` block, and have all of this code on the same base indentation.